### ISSUE TITLE
Package request: rustlings

### ISSUE THREAD

**Issue by StayBlue** [NixOS/nixpkgs#334595]
**Project description**

This project is used by programmers looking to explore the Rust programming language and learn more about it while doing exercises. These exercises are meant to get you comfortable with the language itself and its features.

**Metadata**

* homepage URL: https://rustlings.cool/
* source URL: https://github.com/rust-lang/rustlings
* license: MIT
* platforms: MacOS, Linux

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by mksafavi** [comment]
Interesting package request. 
But does it make sense given that you need to modify the rustling files in order to use it?
 I personally used the flake.nix that's included in the project. which I think are removed in the new release :(

**Comment by StayBlue** [comment]
> Interesting package request. But does it make sense given that you need to modify the rustling files in order to use it? I personally used the flake.nix that's included in the project. which I think are removed in the new release :(

Take a look at [this issue](https://github.com/rust-lang/rustlings/issues/2054). The maintainer of rustlings explained why the flake.nix was removed.

**Comment by mo8it** [comment]
> > Interesting package request. But does it make sense given that you need to modify the rustling files in order to use it? I personally used the flake.nix that's included in the project. which I think are removed in the new release :(
> 
> Take a look at [this issue](https://github.com/rust-lang/rustlings/issues/2054). The maintainer of rustlings explained why the flake.nix was removed.

TL;DR: All required files are embedded in the binary now. The exercises are dumped on initialization.

I am the new author. If you need any help, don't hesitate to ping me.

Note that a new version was published yesterday.
